src/compiler/moduleSpecifiers.ts

```diff
@@ -107,61 +109,63 @@ import {
 // Used by importFixes, getEditsForFileRename, and declaration emit to synthesize import module specifiers.
 
 const enum RelativePreference { Relative, NonRelative, Shortest, ExternalNonRelative }
-// See UserPreferences#importPathEnding
-const enum Ending { Minimal, Index, JsExtension }
 
 // Processed preferences
 interface Preferences {
     readonly relativePreference: RelativePreference;
-    readonly ending: Ending;
+    /**
+     * @param syntaxImpliedNodeFormat Used when the import syntax implies ESM or CJS irrespective of the mode of the file.
+     */
+    getAllowedEndingsInPrefererredOrder(syntaxImpliedNodeFormat?: SourceFile["impliedNodeFormat"]): ModuleSpecifierEnding[];
 }
 
-function getPreferences(host: ModuleSpecifierResolutionHost, { importModuleSpecifierPreference, importModuleSpecifierEnding }: UserPreferences, compilerOptions: CompilerOptions, importingSourceFile: SourceFile): Preferences {

```

Comment:
why doesn't this need `host` anymore? Was it previously reading the file contents? Checking some other preferences?